frontend/src/components/FilterBar.tsx

```diff
@@ -0,0 +1,308 @@
+  filter: (filter: ApiFilter) => void;
+  filterTypes: string[];
+}
+
+interface FilterBarState {
+  filterChips: FilterChip[];
+  lastSelectedFilterType: string;
+  shouldSuggestTypes: boolean;
+  suggestions: string[];
+  textFieldInput: string;
+}
+
+export default class FilterBar extends React.Component<FilterBarProps, FilterBarState> {
+  private _isMounted = true;
+
+  // TODO: remove these once suggestion fetching is implemented
+  private FILTER_VALUE_SUGGESTIONS = ['My Pipeline', 'Pipeline 1', 'Pipeline 2'];
+
+  constructor(props: FilterBarProps) {
+    super(props);
+
+    this.state = {
+      filterChips: [],
+      lastSelectedFilterType: '',
+      // This will control whether we are suggesting the filter types (name, timestamp, etc.) or
+      // actual values
+      shouldSuggestTypes: true,
+      suggestions: props.filterTypes,
+      textFieldInput: '',
+    };
+  }
+
+  public componentWillUnmount(): void {
+    this._isMounted = false;
+  }
+
+  public render(): JSX.Element {
+    const { filterChips, suggestions, textFieldInput } = this.state;
+    return (
+      <React.Fragment>
+        <Autosuggest
+          getSuggestionValue={(suggestion) => suggestion}
+          inputProps={{
+            cancelBubble: true,
+            chips: filterChips,
+            classes,
+            onAdd: this._handleAddChip.bind(this),
+            onChange: this._handleTextFieldInputChange.bind(this),
+            onDelete: this._handleDeleteChip.bind(this),
+            value: textFieldInput,
+          }}
+          onSuggestionHighlighted={this._handleSuggestionHighlighted.bind(this)}
+          onSuggestionSelected={this._onSuggestionSelected.bind(this)}
+          onSuggestionsClearRequested={this._handleSuggestionsClearRequested.bind(this)}
+          onSuggestionsFetchRequested={this._handleSuggestionsFetchRequested.bind(this)}
+          renderInputComponent={this._renderChipInput.bind(this)}
+          renderSuggestion={this._renderSuggestion.bind(this)}
+          renderSuggestionsContainer={this._renderSuggestionsContainer.bind(this)}
+          shouldRenderSuggestions={this._shouldRenderSuggestion.bind(this)}
+          suggestions={suggestions}
+        />
+      </React.Fragment>
+    );
+  }
+
+  private _renderChipInput(inputProps: any): JSX.Element {
+    // TODO we need to pull 'classes' out of the props so it doesn't get passed down, but it
+    // conflicts with the 'classes' from typestyle
+    // tslint:disable-next-line:no-shadowed-variable
+    const { classes, value, cancelBubble, onChange, onAdd, onDelete, ref, ...other } = inputProps;
+
+    return (
+    <ChipInput
+      classes={{}}
+      // value here is the key to updating what is displayed in the bar.
+      InputProps={{ value }}
+      value={this.state.filterChips}
+      onUpdateInput={onChange}
+      onAdd={onAdd}
+      onDelete={onDelete}
+      // this ref is an empty function, but seems to be necessary
+      inputRef={ref}
```

Comment:
That's interesting, refs should always be optional since you might not be using it anywhere. The component complains about this?